website/blog/2019-08-21-announcing-riff-0-4-0.md

```diff
@@ -0,0 +1,124 @@
+
+See the [Core runtime docs](/docs/v0.4/runtimes/core) for details.
+
+#### Knative runtime
+
+The Knative runtime is most analogous to riff 0.3. It requires that Knative Serving and Istio are installed into the cluster in addition to riff. There are two models for consuming Knative: Deployers and Adapters.
+
+Deployers create a Knative Configuration and Route from the latest image available for a referenced build. The Configuration is updated as the build produces new images.
+
+Adapters reference an existing Knative Service or Configuration, updating the image. The Route rules are preserved as new images trigger the creation of Knative Revisions.
+
+See the [Knative runtime docs](/docs/v0.4/runtimes/knative) for details.
+
+#### Streaming runtime
+
+The new streaming runtime is under active development, but not ready in time for the v0.4 release.
+
+<!-- TODO say more about what makes our streaming model unique -->
+
+Development can be tracked on GitHub, including:
+- [streaming-processor](https://github.com/projectriff/streaming-processor)
+- [kafka-provider](https://github.com/projectriff/kafka-provider)
+- [streaming-http-adapter](https://github.com/projectriff/streaming-http-adapter)
+
+### riff CLI
+
+The [riff CLI](/docs/v0.4/cli/riff) was rewritten from the ground up to support the new riff Build and Runtime models. The CLI behavior is more consistent and predictable. No more deleting a service that was created as a function.
+
+The most significant change is that creating a function only builds source into a container image, it no longer also deploys the function as a Knative workload. Knative is just one runtime option.
+
+A few of the key highlights include:
+- table `list`ings with rolled up status
+- colorized output
+- `--tail` to watch logs during `create` until the resource is ready or fails
+- `tail` command to watch all logs for a resource until canceled
+- `delete` and `list` for every resource that is `create`d
+
+riff is no longer installed into a Kubernetes cluster using the CLI. [Helm charts](#helm-charts) are available to aid the installation, we are exploring other installation options, like [Cloud Native Application Bundles](https://cnab.io) (CNAB).
+
+### riff System and CRDs
+
+Powering the new build and runtime models is riff System. riff System provides Kubernetes CRDs and a controller to reconcile the state of these projectriff resources into other resources to achieve the desired outcome.
+
+The system provides four API groups, one for builds and one per runtime:
+
+- `build.projectriff.io/v1alpha1`
+  - `Application` - applications built from source using application buildpacks
+  - `Function` - functions built from source using function buildpacks
+  - `Container` - watch a container repository for the latest image
+- `core.projectriff.io/v1alpha1`
+  - `Deployer` - deployers map HTTP requests to applications, functions, containers or images with Kubernetes core resources
```

Comment:
the Deployer is really just a reconciler of declarative resources so I don't think we should suggest it is actively _mapping_ http requests

for that reason (and consistency), I think the Knative "Adapter" description below is the right example, so this could change to:

"deployers map applications, functions, or containers or images to Kubernetes core resources: Deployment and Service" ... maybe with an added point that it can then handle HTTP requests via that Service